src/LogFileReader.cpp

```diff
@@ -74,12 +74,19 @@ bool LogFileReader::readAll(const QString& fileName)
 // File input is a csv file in the format of hh:mm:ss:zzz, voltage, current
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
-    // TODO implement this first
-
-    // This is here to the compiler happy. Otherwise the compile
-    // will have an error warning about an unused variable. Remove this
-    // when you use it.
-    Q_UNUSED(line);
-    Q_UNUSED(batteryData);
+    QString::SectionFlag flag = QString::SectionSkipEmpty;
```

Comment:
I would be wary of this line, since skipping sections that are empty could be dangerous since you know exactly what the first second and third sections should be. If there is an empty section, there is a problem and you should know and indicate that there is an issue.